### PR TITLE
[WIP] Region vid newtype index

### PR DESCRIPTION
**Pull request by spastorino**
THIS SHOULD BE MERGED AFTER https://github.com/rust-lang/rust/pull/45825 and probably after rebasing and force pushing :)
/cc @nikomatsakis

### CODE REVIEW

**Comment by shepmaster** [comment]
Auto-assigning to...

r? @estebank

**Comment by nikomatsakis** [comment]
Note the [new policy on landing things for NLL](https://github.com/rust-lang/rust/issues/43234#issuecomment-343523271) -- basically, use the `nll-master` branch on my fork for now, and I will take care of the rest. @spastorino, I'm going to close this PR, can you re-open it against my repository?

File: src/librustc/infer/region_constraints/mod.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@ use self::CombineMapType::*;
 use super::{MiscVariable, RegionVariableOrigin, SubregionOrigin};
 use super::unify_key;
 
-use rustc_data_structures::indexed_vec::IndexVec;
+use rustc_data_structures::indexed_vec::{IndexVec, Idx};
```
Comment (estebank) [inline src/librustc/infer/region_constraints/mod.rs]: Stray import?

**Review by estebank — COMMENTED** [review]
Looking only at cf7fe00d9181240a0e9aee8e33294817b3625bf8, LGTM. Please resubmit the PR against https://github.com/nikomatsakis/rust/tree/nll-master

File: src/librustc/infer/region_constraints/mod.rs (hunk 1 above)
Comment (spastorino) [reply src/librustc/infer/region_constraints/mod.rs]: It's needed because methods of Idx trait are used

Comment (estebank) [reply src/librustc/infer/region_constraints/mod.rs]: Ah, I see.

**Comment by spastorino** [comment]
Done here https://github.com/nikomatsakis/rust/pull/11

**Pull request closed without merging**